File: PetFinder_Android/Pet Adopt App/PetAdoptApp/app/src/main/java/com/nati/petadoptapp/data/pet/PetsDataImpl.kt
package com.nati.petadoptapp.data.pet

import com.nati.petadoptapp.data.pet.remote.PetsRemoteImpl
import com.nati.petadoptapp.domain.PetsRepository
import com.nati.petadoptapp.model.Organization
import com.nati.petadoptapp.model.Pet
import com.nati.petadoptapp.model.user.PetFireStore

class PetsDataImpl(
    private val petRemoteImpl: PetsRemoteImpl
) : PetsRepository {

    override suspend fun getPets(type: String): List<Pet> {
        return petRemoteImpl.getPets(type)
    }

    /*override suspend fun getPets(): List<Pet> {
        return petRemoteImpl.getPets()
    }*/

    override suspend fun getPet(petId: Int): Pet {
        return petRemoteImpl.getPet(petId)
    }

    override suspend fun getOrganizations(): List<Organization> {
        return petRemoteImpl.getOrganizations()
    }

    override suspend fun getOrganization(organizationId: String): Organization {
        return petRemoteImpl.getOrganization(organizationId)
    }

    override suspend fun addPet(pet: PetFireStore): String {
        return petRemoteImpl.addPet(pet)
    }
}

File: PetFinder_Flutter/pet_adopt_app/lib/data/pets/pets_data_impl.dart
import 'package:pet_adopt_app/data/pets/local/pets_local_impl.dart';
import 'package:pet_adopt_app/data/pets/remote/mapper/pets_remote_mapper.dart';
import 'package:pet_adopt_app/data/pets/remote/pets_remote_impl.dart';
import 'package:pet_adopt_app/data/token/local/token_local_impl.dart';
import 'package:pet_adopt_app/domain/pets_repository.dart';
import 'package:pet_adopt_app/model/pets/pets.dart';

class PetsDataImpl extends PetsRepository {
  final PetsRemoteImpl _remoteImpl;
  final PetsLocalImpl _localImpl;
  final TokenLocalImpl _tokenLocalImpl;

  PetsDataImpl(
      {required PetsRemoteImpl remoteImpl,
      required PetsLocalImpl localImpl,
      required TokenLocalImpl tokenLocalImpl})
      : _remoteImpl = remoteImpl,
        _tokenLocalImpl = tokenLocalImpl,
        _localImpl = localImpl;

  @override
  Future<PetsNetworkResponse> getPets(String type, String page) async {
    final token = await _tokenLocalImpl.loadToken();
    final remotePets = await _remoteImpl.getPets(token, type, page);
    return PetsRemoteMapper.fromRemote(remotePets);
  }

  @override
  Future<Pet> getPet(int petId) async {
    final token = await _tokenLocalImpl.loadToken();
    return await _remoteImpl.getPet(token, petId);
  }

  @override
  Future<List<PetType>> getPetTypes() async {
    final token = await _tokenLocalImpl.loadToken();
    return await _remoteImpl.getPetTypes(token);
  }

  @override
  Future<void> addPetFavorite(Pet pet) async {
    await _localImpl.addPetFavorites(pet);
  }

  @override
  Future<void> deletePetFavorite(Pet pet) async {
    await _localImpl.deletePetFavorites(pet);
  }

  @override
  Future<List<Pet>> getPetsFavorites() async {
    return await _localImpl.getPetFavorites();
  }

  @override
  Future<bool> isFavorite(Pet pet) async {
    return await _localImpl.isFavorite(pet);
  }
}
